File: backend/internal/model/note.go
package model

import (
	"time"

	"github.com/google/uuid"
)

// Note represents a user's note
type Note struct {
	ID            uuid.UUID  `json:"id"`
	UserID        uuid.UUID  `json:"user_id"`
	TransactionID *uuid.UUID `json:"transaction_id,omitempty"`
	Title         string     `json:"title"`
	Content       string     `json:"content"`
	Color         string     `json:"color,omitempty"`
	IsPinned      bool       `json:"is_pinned"`
	CreatedAt     time.Time  `json:"created_at"`
	UpdatedAt     time.Time  `json:"updated_at"`
}

// CreateNoteRequest represents a request to create a note
type CreateNoteRequest struct {
	Title         string  `json:"title"`
	Content       string  `json:"content"`
	Color         string  `json:"color,omitempty"`
	IsPinned      bool    `json:"is_pinned,omitempty"`
	TransactionID *string `json:"transaction_id,omitempty"`
}

// UpdateNoteRequest represents a request to update a note
type UpdateNoteRequest struct {
	Title         *string `json:"title,omitempty"`
	Content       *string `json:"content,omitempty"`
	Color         *string `json:"color,omitempty"`
	IsPinned      *bool   `json:"is_pinned,omitempty"`
	TransactionID *string `json:"transaction_id,omitempty"`
}

// NoteColors represents available note colors
var NoteColors = []string{
	"default",
	"red",
	"orange",
	"yellow",
	"green",
	"blue",
	"purple",
	"pink",
}

File: app/src/types/note.ts
// Note types
export interface Note {
  id: string;
  user_id: string;
  transaction_id?: string;
  title: string;
  content: string;
  color: string;
  is_pinned: boolean;
  created_at: string;
  updated_at: string;
}

export interface CreateNoteRequest {
  title: string;
  content?: string;
  color?: string;
  is_pinned?: boolean;
  transaction_id?: string;
}

export interface UpdateNoteRequest {
  title?: string;
  content?: string;
  color?: string;
  is_pinned?: boolean;
  transaction_id?: string;
}

export const NOTE_COLORS = [
  'default',
  'red',
  'orange',
  'yellow',
  'green',
  'blue',
  'purple',
  'pink',
] as const;

export type NoteColor = (typeof NOTE_COLORS)[number];
